### DISCUSSION TITLE
Is it Safe to Overwrite `DagBag` in Airflow with a Plugin?

### DISCUSSION THREAD

**Discussion by Rmjp** [apache/airflow#39832] [Q&A] [answered]
I want `DagBag` to collect only specific files, so I created a custom DagBag class that modifies the `process_file` function. I then overwrote the default `DagBag` in the `on_load` function of my plugin.

Is this approach safe? Are there any considerations or potential issues I should be aware of?

```
class CustomDagBag(DagBag):
    def __init__(self, *args, **kwargs):
        self.suffix = os.getenv("DAG_FILE_SUFFIX", "")
        super().__init__(*args, **kwargs)


    def process_file(self, filepath, only_if_updated=True, safe_mode=True):
        suffix = self.suffix
        if re.match(r".*-{}.*".format(suffix), filepath) is None:
            return []
        dags = super().process_file(filepath, only_if_updated, safe_mode)
        return dags
```

```
class CustomDagLoaderPlugin(AirflowPlugin):
    name = "custom_dag_loader"
    ...
    
    def on_load(self):
        airflow.models.dagbag.DagBag = CustomDagBag
        pass
```

**Comment by josh-fell** [comment, marked as answer]
Have you checked out [using a `.airflowignore` file](https://airflow.apache.org/docs/apache-airflow/stable/core-concepts/dags.html#airflowignore) to control what DAG files are collected?

**Reply by Rmjp** [reply]
thank, i will do that.
